### DISCUSSION TITLE
Update all fields to lowercase

### DISCUSSION THREAD

**Discussion by masclot** [firefly-iii/firefly-iii#5969] [Firefly III]
Hi,
I'm using the data importer for my bank statements. Unfortunately each bank chooses a different format for the text: all uppercase, lowercase, camelcase,... This is making searches very difficult and my rules have many redundancies. I decided to modify my bank statements to be all lowercase before importing. However, there is quite some data in the tool. Instead of deleting it all and start over, I wonder if it is possible to just go directly to the database and modify the fields to be lowercase. The description field should be enough. I' m using PostgreSQL.
If it is possible, which tables and fields should I change?
Thanks!

**Comment by JC5** [comment]
Perhaps it would be easier to migrate to MySQL?

**Comment by masclot** [comment]
I'm using Raspberry Pi + docker, a combination not supported by MySql.

**Reply by JC5** [reply]
There are ARM images of mariaDB.

**Comment by masclot** [comment]
I didn't know about mariaDb, thanks. I still need to run scripts to clean
the output of the bank statements, before importing them, as they are not
100% cvs. Adding a lowercase conversion in my scripts is very easy. Does
mariaDb work better than PostgreSql in any other way to justify migrating
to it? Importing the transactions again is not an issue, but recreating
tags, rules and categories would be very time consuming.

**Reply by JC5** [reply]
I don't know, I was just offering an alternative. MySQL works perfectly well for me. There are scripts online that can convert between MySQL and PostgreSQL and vice-versa, but if making everything lowercase is just as easy, than it's an easy choice.

**Comment by skorvek** [comment]
Perhaps a preference for case-insensitive searches?  I use Postgresql, and am not at all interested in switching o MySQL. I don't feel that the datatbase choice is relevant in whether searches are case sensitive or case insensitive.

I can see where a rule criteria may need to be case-sensitive but if I'm searching transactions using the top-left search bar, I'd generally expect to find that text string regardless of case. I'm often searching again when I don't find what I expect because- as per the original post- the import data had a different case.

**Reply by JC5** [reply]
> I don't feel that the datatbase choice is relevant in whether searches are case sensitive or case insensitive.

It's not, this entirely on PostgreSQL.
